### PR TITLE
Fix: align send micro examples

### LINKED ISSUE #126: Align Send a Transaction and Send a Micro Transaction examples

### LINKED ISSUE #234: [Bug]: sendAmount() breaks in the examples
### Issue description

Since the merge(?) of sendAmount() and sendMicroAmount() in https://github.com/iotaledger/iota-sdk/pull/205, I cannot run the nodejs `7-events.js` example anymore

### Version

`develop`

### Expected behaviour

7-events.js runs without problems

### Actual behaviour

Error is thrown

### Can the issue reliably be reproduced?

Yes

### Steps to reproduce the issue

Run `node 7-events.js` from the `sdk/src/wallet/bindings/nodejs/examples` folder.

### Errors

Error:  {
  type: 'json',
  error: '`missing field `address` at line 1 column 232`'
}

### Duplicate declaration

- [X] I have searched the issues tracker this issue and there is none

### PR DESCRIPTION
**Pull request by kwek20**
# Description of change

Aligns examples for send amount and micro

Still needs python env updates as this did not exist yet
See: https://github.com/iotaledger/iota-sdk/issues/240

## Links to any relevant issues

Fixes https://github.com/iotaledger/iota-sdk/issues/126
Fixes https://github.com/iotaledger/iota-sdk/issues/234

## Type of change
- Enhancement (a non-breaking change which adds functionality)

## How the change has been tested

ran the tests

## Change checklist

Tick the boxes that are relevant to your changes, and delete any items that are not.

- [ ] I have followed the contribution guidelines for this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that new and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by Thoralf-M — DISMISSED** [review]

File: sdk/Cargo.toml [hunk 1]
```diff
@@ -72,6 +72,7 @@ fern-logger = { version = "0.5.0", default-features = false }
 [features]
 client = [ "serde", "block", "dto", "api", "pow", "std", "async-trait", "log", "derive_builder" ]
 pow = [ ]
+rocks_db = [ "rocksdb", "storage" ]
```
Comment (thibault-martinez) [inline sdk/Cargo.toml]:
```suggestion
rocksdb = [ "dep:rocksdb", "storage" ]
```

File: sdk/examples/wallet/06_send_micro_transaction.rs [hunk 2]
```diff
@@ -24,6 +24,7 @@ async fn main() -> Result<()> {
         .set_stronghold_password(&std::env::var("STRONGHOLD_PASSWORD").unwrap())
         .await?;
 
+    // Send a micro transaction with 1 i
```
Comment (thibault-martinez) [inline sdk/examples/wallet/06_send_micro_transaction.rs]: Hmm don't put `i`

File: sdk/src/wallet/bindings/java/examples/src/BurnNativeToken.java [hunk 3]
```diff
@@ -31,13 +31,16 @@ public static void main(String[] args) throws WalletException, InterruptedExcept
...
                 .withTokenId(tokenId)
                 .withBurnAmount(burnAmount)
         );
 
         // Print transaction.
-        System.out.println(t);
+        System.out.println(
+                "Transaction: " + transaction.getTransactionId() +
+                        " Block sent: " + Env.NODE + "/api/core/v2/blocks/" +
+                        transaction.getBlockId());
```
Comment (thibault-martinez) [inline sdk/src/wallet/bindings/java/examples/src/BurnNativeToken.java]: Am I too annoying if I ask you to print them in 2 different lines ?

File: sdk/Cargo.toml (hunk 1 above)
Comment (kwek20) [reply sdk/Cargo.toml]: oh thats how to do it, fancy

**Review by thibault-martinez — CHANGES_REQUESTED** [review]
Conflicts

File: sdk/examples/wallet/06_send_micro_transaction.rs [hunk 4]
```diff
@@ -24,6 +24,7 @@ async fn main() -> Result<()> {
         .set_stronghold_password(&std::env::var("STRONGHOLD_PASSWORD").unwrap())
         .await?;
 
+    // Send a micro transaction with 1
```
Comment (thibault-martinez) [inline sdk/examples/wallet/06_send_micro_transaction.rs]:
```suggestion
    // Send a micro transaction with amount 1
```

File: sdk/src/wallet/bindings/java/examples/src/SendNativeToken.java [hunk 5]
```diff
@@ -32,15 +32,19 @@ public static void main(String[] args) throws WalletException, InterruptedExcept
...
+                                                        new AddressNativeTokens.NativeTokenTuple(tokenId, tokenAmount)
+                                        })
+                        }));
 
         // Print transaction.
-        System.out.println(t);
+        System.out.println(
+                        "Transaction: " + transaction.getTransactionId() +
+                                        " Block sent: " + Env.NODE + "/api/core/v2/blocks/" +
+                                        transaction.getBlockId());
```
Comment (thibault-martinez) [inline sdk/src/wallet/bindings/java/examples/src/SendNativeToken.java]: Split in two

**Comment by thibault-martinez** [comment]
What was `Remove funding the address.` about? Because I don't see it being changed in this PR?

Also, python doesn't need any alignement?
And nodejs didn't need much as well?

File: sdk/src/wallet/bindings/java/examples/src/SendAmount.java [hunk 6]
```diff
@@ -20,25 +23,26 @@ public static void main(String[] args) throws WalletException, InterruptedExcept
...
-        }));
+        Transaction transaction = account.sendAmount(new org.iota.types.account_methods.SendAmount()
+                        .withAddressesWithAmount(outputs));
 
         // Print transaction.
-        System.out.println(t);
+        System.out.println(
+                        "Transaction: " + transaction.getTransactionId() +
+                                        " Block sent: " + Env.NODE + "/api/core/v2/blocks/" +
+                                        transaction.getBlockId());
```
Comment (thibault-martinez) [inline sdk/src/wallet/bindings/java/examples/src/SendAmount.java]: Split

File: sdk/src/wallet/bindings/java/examples/src/Env.java [hunk 7]
```diff
@@ -2,8 +2,9 @@
 // SPDX-License-Identifier: Apache-2.0
 
 public class Env {
-    public static final String NODE = "https://api.testnet.shimmer.network";
-    public static final String EXPLORER = "explorer.shimmer.network/testnet";
+    private static string NETWORK = "testnet";
+    public static final String NODE = "https://api." + NETWORK + ".shimmer.network";
```
Comment (Thoralf-M) [inline sdk/src/wallet/bindings/java/examples/src/Env.java]: Just do
```suggestion
    public static final String NODE = "https://api.testnet.shimmer.network";
```
For shimmer it's `https://api.shimmer.network` and not `staging` or `mainnet`, so this network separation doesn't really make sense

Comment (kwek20) [reply sdk/src/wallet/bindings/java/examples/src/Env.java]: ah thats too bad :(

File: sdk/src/wallet/bindings/java/examples/src/Env.java [hunk 8]
```diff
@@ -2,9 +2,8 @@
 // SPDX-License-Identifier: Apache-2.0
 
 public class Env {
-    private static string NETWORK = "testnet";
-    public static final String NODE = "https://api." + NETWORK + ".shimmer.network";
-    public static final String EXPLORER = "explorer.shimmer.network/" + NETWORK;
+    public static final String NODE = "https://api.testnet.shimmer.network";
+    public static final String EXPLORER = "explorer.shimmer.network/testnet";
```
Comment (Thoralf-M) [inline sdk/src/wallet/bindings/java/examples/src/Env.java]:
```suggestion
    public static final String EXPLORER = "https://explorer.shimmer.network/testnet";
```
not needed?

**Comment by thibault-martinez** [comment]
Conflicts

File: sdk/src/wallet/bindings/python/examples/3-send-transaction.py [hunk 9]
```diff
@@ -2,6 +2,9 @@
 
 # This example sends a transaction.
 
+# Shimmer coin type
+EXPLORER = ""
```
Comment (Thoralf-M) [inline sdk/src/wallet/bindings/python/examples/3-send-transaction.py]: What's going on here? :trollface:

File: sdk/src/wallet/bindings/nodejs/examples/.env.example [hunk 10]
```diff
@@ -1,4 +1,5 @@
 STRONGHOLD_PASSWORD="A12345678*"
 MNEMONIC="bulk spoon broken license diary nominee tribe visit used giant rail insect lesson home toast autumn cancel alley park give same wet wash vanish"
 NODE_URL="https://api.testnet.shimmer.network"
-FAUCET_URL="https://faucet.testnet.shimmer.network/api/enqueue"
\ No newline at end of file
+FAUCET_URL="https://faucet.testnet.shimmer.network/api/enqueue"
+EXPLORER_URL = "explorer.shimmer.network/testnet"
```
Comment (thibault-martinez) [inline sdk/src/wallet/bindings/nodejs/examples/.env.example]: https?

File: sdk/src/wallet/bindings/python/examples/4-send-micro-transaction.py [hunk 11]
```diff
@@ -19,4 +19,5 @@
 
 transaction = account.send_amount(outputs, { "allowMicroAmount": True })
 
-print(f'Sent transaction: {transaction}')
+print(f'Transaction: {transaction.transaction_id}')
+print(f'Block sent: {EXPLORER}/block/" + {transaction.block_id});
```
Comment (thibault-martinez) [inline sdk/src/wallet/bindings/python/examples/4-send-micro-transaction.py:26]: Missing constant?

File: sdk/src/wallet/bindings/python/examples/6-send-nft.py [hunk 12]
```diff
@@ -19,4 +19,5 @@
 
 transaction = account.send_nft(outputs)
 
-print(f'Sent transaction: {transaction}')
+print(f'Transaction: {transaction.transaction_id}')
+print(f'Block sent: {EXPLORER}/block/" + {transaction.block_id});
```
Comment (thibault-martinez) [inline sdk/src/wallet/bindings/python/examples/6-send-nft.py:26]: same

File: sdk/src/wallet/bindings/python/examples/5-send-native-tokens.py [hunk 13]
```diff
@@ -23,4 +23,5 @@
 
 transaction = account.send_native_tokens(outputs, None)
 
-print(f'Sent transaction: {transaction}')
+print(f'Transaction: {transaction.transaction_id}')
+print(f'Block sent: {EXPLORER}/block/" + {transaction.block_id});
```
Comment (thibault-martinez) [inline sdk/src/wallet/bindings/python/examples/5-send-native-tokens.py:30]: same

**Comment by thibault-martinez** [comment]
What was `Remove funding the address.` ?

**Review by thibault-martinez — DISMISSED** [review]

File: sdk/src/wallet/bindings/java/examples/src/RequestFundsFromFaucet.java [hunk 14]
```diff
@@ -22,6 +22,8 @@ public static void main(String[] args) throws WalletException, InterruptedExcept
                 .withStoragePath(Env.STORAGE_PATH)
         );
 
+        wallet.setStrongholdPassword(Env.STRONGHOLD_PASSWORD);
```
Comment (DaughterOfMars) [inline sdk/src/wallet/bindings/java/examples/src/RequestFundsFromFaucet.java]: Where's this change coming from?

File: sdk/src/wallet/bindings/nodejs/examples/4a-send-amount.js [hunk 15]
```diff
@@ -8,9 +8,6 @@ async function run() {
     try {
         const manager = await getUnlockedManager();
         const account = await manager.getAccount('Alice');
-        console.log('Account:', account);
-
-        await account.sync();
```
Comment (DaughterOfMars) [inline sdk/src/wallet/bindings/nodejs/examples/4a-send-amount.js:13]: Is this not needed?

File: sdk/src/wallet/bindings/java/examples/src/RequestFundsFromFaucet.java (hunk 14 above)
Comment (kwek20) [reply sdk/src/wallet/bindings/java/examples/src/RequestFundsFromFaucet.java]: removed

Comment (DaughterOfMars) [reply sdk/src/wallet/bindings/java/examples/src/RequestFundsFromFaucet.java]: There were a few instances I noted. Are any of them supposed to be there?

Comment (kwek20) [reply sdk/src/wallet/bindings/java/examples/src/RequestFundsFromFaucet.java]: not in java for sure

File: sdk/src/wallet/bindings/nodejs/examples/4a-send-amount.js (hunk 15 above)
Comment (kwek20) [reply sdk/src/wallet/bindings/nodejs/examples/4a-send-amount.js:13]: if you follow the steps, whatever is needed should be stored locally

Comment (DaughterOfMars) [reply sdk/src/wallet/bindings/nodejs/examples/4a-send-amount.js:13]: Sure, but I added sync calls intentionally to illustrate their usage.

**Review by thibault-martinez — DISMISSED** [review]

File: sdk/src/wallet/bindings/nodejs/examples/4c-send-micro-transaction.js [hunk 16]
```diff
@@ -9,12 +9,10 @@ async function run() {
         const manager = await getUnlockedManager();
         const account = await manager.getAccount('Alice');
```
Comment (Thoralf-M) [inline sdk/src/wallet/bindings/nodejs/examples/4c-send-micro-transaction.js]:
```suggestion

        await account.sync();

```

File: sdk/src/wallet/bindings/nodejs/examples/4a-send-amount.js [hunk 17]
```diff
@@ -8,9 +8,6 @@ async function run() {
     try {
         const manager = await getUnlockedManager();
         const account = await manager.getAccount('Alice');
```
Comment (Thoralf-M) [inline sdk/src/wallet/bindings/nodejs/examples/4a-send-amount.js:10]:
```suggestion
        const account = await manager.getAccount('Alice');

        await account.sync();
```

**Review by Thoralf-M — APPROVED** [review]

**Review by DaughterOfMars — APPROVED** [review]

**Review by thibault-martinez — APPROVED** [review]

**Pull request merged**